_clanky/znalostní-grafy-02-rdf.md

```diff
@@ -0,0 +1,339 @@
+
+{% include image.html 
+   url="../attachments/články/znalostní-grafy/rdf-čsú-04.svg"
+   description="Snadné slévání RDF trojic z různých datových zdrojů"
+%}
+
+Na obrázku můžete také vidět, že znalostní graf Registru práv a povinností specifikuje adresní místo sídla ČSÚ. Adresní místo chápe jako věc, pro níž používá IRI z domény Českého úřadu zeměměřičského a katastrálního (ČÚZK). Fialová hrana na obrázku značí RDF trojici ze znalostního grafu RÚIAN, který spravuje ČÚZK. Na obrázku tak vidíme trojice ze tří různých znalostních grafů. Abychom s nimi mohli pracovat, nemusíme dělat nic víc, než je slít na jedno místo.
+
+Výše uvedená situace ale neodpovídá skutečnosti. Ve skutečnosti není vhodné ve znalostním grafu Registru práv a povinností používat k identifikaci ČSÚ IRI ze znalostního grafu Národního katalogu otevřených dat. Jde např. o dereferencování. Prostřednictvím dereferencování tohoto IRI se vždy dostaneme pouze na tvrzení ze znalostního grafu Národního katalogu otevřených dat, nikoliv na tvrzení ze znalostního grafu Registru práv a povinností. Proto používá znalostní graf Registru práv a povinností pro identifikaci ČSÚ vlastní IRI [`https://rpp-opendata.egon.gov.cz/odrpp/zdroj/orgán-veřejné-moci/00025593`](https://rpp-opendata.egon.gov.cz/odrpp/zdroj/orgán-veřejné-moci/00025593). Tím ale přicházíme o výhodu popsanou výše. Jedná se o běžnou situaci, kdy různé datové zdroje identifikují stejnou věc svými vlastními a tudíž různými identifikátory. Většina datových modelů a jazyků řeší tuto situaci obdobou klíčů a cizích klíčů. Někde ve schématu, které je vyjádřeno jinde, v jiném jazyku a v jiném modelu, je popsáno, jakým způsobem se na sebe reprezentace věcí odkazují. Přímo v datech tato informace obsažena není. V modelu RDF je naopak možné ji zaznamenat přímo jako součást dat stejně jako jakékoliv jiné tvrzení a to v podobě trojice s predikátem, který je k tomu přímo určený. Jedná se o predikát [`http://www.w3.org/2002/07/owl#sameAs`](http://www.w3.org/2002/07/owl#sameAs) ze slovníku OWL. Pomocí něj vyjádříme, že dva různé uzly identifikované různými IRI ve skutečnosti reprezentují tutéž věc. Situaci, kdy oba znalostní grafy identifikují ČSÚ jiným IRI ukazuje následující obrázek. Trojice vyjadřující, že obě IRI reprezentují ČSÚ, je součástí znalostního grafu Národního katalogu otevřených dat.
+
+{% include image.html 
+   url="../attachments/články/znalostní-grafy/rdf-čsú-05.svg"
+   description="Využití vlatnosti [`http://www.w3.org/2002/07/owl#sameAs`](http://www.w3.org/2002/07/owl#sameAs) ze slovníku OWL pro propojování znalostních grafů"
+%}
```

Comment:
Tady taky chybí uvozovka a language tag u názvu ČSÚ.

Navíc bych ale owl:sameAs v grafu udělal třeba červeně. Těch hran už je hodně, a když někdo neví kam koukat, tak to tam možná nenajde.

Reply:
Já o tom sameAs mluvím ještě dál, že se může stát, že ho musí dodělat někdo další a tam to barevně zvýrazňuju. Už mi to přijde až moc zdůrazňovaný. Tak jsem to jenom ztučnil.